src/output-json.js

```diff
@@ -12,12 +15,34 @@ const outputJson = (
     jsonItems: Array<JsonItem>,
     violationFileNames: Array<string>,
 ): ErrorCode => {
+    const regex = new RegExp(escapeRegExp(path.sep), "g");
```

Comment:
So as we discussed, I expect the correct fix is to modify the StringLogger to cope with the double escapes. This change will create invalid JSON for windows output, I think.